provider/github/watcher.go

```diff
@@ -146,23 +234,44 @@ func (w *Watcher) doEventRequest(ctx context.Context, username, repository strin
 		return resp, nil, err
 	}
 
+	interval := w.newInterval(resp)
+	// obey poll interval
 	secs, _ := strconv.Atoi(resp.Response.Header.Get("X-Poll-Interval"))
-	w.pollInterval = time.Duration(secs) * time.Second
+	pollLimit := time.Duration(secs) * time.Second
+	if pollLimit > interval {
+		interval = pollLimit
+	}
+	w.eventsPollInterval = interval
 
 	if isStatusNotModified(resp.Response) {
 		return nil, nil, NoErrNotModified.New()
 	}
 
-	log.With(log.Fields{
-		"remaining-requests": resp.Rate.Remaining,
-		"reset-at":           resp.Rate.Reset,
-		"poll-interval":      w.pollInterval,
-		"events":             len(events),
-	}).Debugf("Request to events endpoint done.")
+	w.responseLogger(resp).With(log.Fields{"poll-interval": w.eventsPollInterval}).
+		Debugf("Request to events endpoint done with %d events.", len(events))
 
 	return resp, events, err
 }
 
+func (w *Watcher) newInterval(resp *github.Response) time.Duration {
+	remaining := resp.Rate.Remaining / 2 // we call 2 endpoints for each repo
```

Comment:
What happens when remaining reaches 0? We should wait until `resp.Rate.Reset`, but it looks like that won't be the case in this method.

Reply:
We will try to pull every `minInterval`. Github will start rejecting our calls. For the scope of this PR it's okay I think. But I need a check for zero 👍